src/compile/data/aggregate.ts

```diff
@@ -0,0 +1,134 @@
+import {AggregateOp} from '../../aggregate';
+import {Channel} from '../../channel';
+import {SOURCE} from '../../data';
+import {field, FieldDef} from '../../fielddef';
+import {keys, vals, reduce, hash, Dict, StringSet, allSame, empty} from '../../util';
+import {VgData} from '../../vega.schema';
+
+import {FacetModel} from './../facet';
+import {LayerModel} from './../layer';
+import {Model} from './../model';
+
+import {DataComponent, AggregateComponent} from './data';
+
+
+export namespace aggregate {
+  function addDimension(dims: { [field: string]: boolean }, fieldDef: FieldDef) {
+    if (fieldDef.bin) {
+      dims[field(fieldDef, { binSuffix: '_start' })] = true;
```

Comment:
`// To avoid confusion, group by all types of bin values so we always have them if we have to use`